src/class/ResponseHeader/ResponseHeader.cpp

```diff
@@ -0,0 +1,224 @@
+
+#include <ResponseHeader/ResponseHeader.hpp>
+
+/*
+** ------------------------------- CONSTRUCTOR --------------------------------
+*/
+
+ResponseHeader::ResponseHeader(int code, const ErrorPage &default_page)
+{
+	defaultConstructor();
+	setStatusCode(code);
+	if (code >= 400)
+	{
+		setContentType(".html");
+		setConnection("close");
+		defaultErrorPage(code, default_page);
+		if (code == 405)
+			_header[CONNECTION].name = "Allow: ";
+	}
+}
+
+ResponseHeader::ResponseHeader(const ResponseHeader &src)
+	: _header(src.getHeader())
+{
+}
+
+const char *ResponseHeader::InvalidDefaultPage::what() const throw()
+{
+	return "Invalid default page";
+}
+
+/*
+** -------------------------------- DESTRUCTOR --------------------------------
+*/
+
+ResponseHeader::~ResponseHeader()
+{
+}
+
+/*
+** --------------------------------- OVERLOAD ---------------------------------
+*/
+
+ResponseHeader &ResponseHeader::operator=(ResponseHeader const &rhs)
+{
+	if (this != &rhs)
+	{
+		this->_header = rhs.getHeader();
+	}
+	return *this;
+}
+
+std::ostream &operator<<(std::ostream &o, ResponseHeader const &i)
+{
+	o << i.getResponse();
+	return o;
+}
+
+/*
+** --------------------------------- METHODS ----------------------------------
+*/
+
+void ResponseHeader::defaultConstructor()
+{
+
+	for (field f = STANDARD; f <= BODY; f = static_cast<field>(static_cast<int>(f) + 1))
+	{
+		HeaderFields defaultHeader;
+		_header.insert(std::make_pair(f, defaultHeader));
+	}
+	_header[STANDARD].name = "HTTP/1.1 ";
+	_header[STANDARD].value = "";
+	_header[SERVER].name = "Server: ";
+	_header[SERVER].value = "webserv";
+	_header[DATE].name = "Date: ";
+	_header[DATE].value = getCurrentDate();
+	_header[CONNECTION_TYPE].name = "Content-Type: ";
+	_header[CONNECTION_TYPE].value = "";
+	_header[CONTENT_LENGHT].name = "Content-Length: ";
+	_header[CONTENT_LENGHT].value = "";
+	_header[CONNECTION].name = "Connection: ";
+	_header[BODY].name = "\r\n";
+	_header[BODY].value = "";
+	setContentType(".txt");
+	setContentLength(0);
+	setConnection("keep-alive");
+}
+
+void ResponseHeader::defaultErrorPage(int code, const ErrorPage &default_page)
```

Comment:
the idea behind this func is to setup the correct error page in case where the status code is > 400 right?

if so, then this func needs all of the error pages: `const ErrorPage &default_page` -> `std::list<const IErrorPage *>`

and maybe the name should be `defaultErrorPage` -> `setErrorPageIfNeeded`?

Reply:
if you want we can leave the pointer but with ErrorPage.